src/message.py

```diff
@@ -24,36 +25,53 @@ def __init__(self, code, body=None):
 
     def send_response(self):
         """Format reponse for in http protocol."""
-        message = '{0} {1}{2}{3}'.format(
+        # message = '{0} {1}{2}{3}'.format(
+        #     HTML_PROTOCOL,
+        #     RESPONSE_CODE[self.code],
+        #     CRLF + CRLF,
+        #     self.body)
+        message = '{0} {1}{2}'.format(
             HTML_PROTOCOL,
             RESPONSE_CODE[self.code],
-            CRLF + CRLF,
-            self.body)
-        return message.encode('utf8')
+            CRLF + CRLF)
+        message = message.encode('utf-8')
+        try:
+            message += self.body[1]
+        except TypeError:
+            if self.code == 200:
+                html_list = '<html>'
+                for item in self.body[1]:
+                    html_list += '<li>{0}'.format(item)
+                message += html_list.encode('utf-8')
+            else:
+                message += self.body.encode('utf-8')
+
+        print(message)
+        return message
 
 
 class Request(object):
     """Parses a request and saves information."""
 
     def __init__(self, user_request):
         """Initialize request by parsing info out of the request."""
```

Comment:
This method will not be nearly as flexible and versatile as you would like. What happens if the request has no CRLFs in it? What if it has too many? Or too many items in the first row, or too few?

There are a lot of opportunities for unhandled IndexErrors in lines 58-63; this indicates that your tests for this part of the code are not nearly rigorous enough.